### PR TITLE
Drool unit tests for judicial scenarios scss

### PR DESCRIPTION
**Pull request by vladkovalskyi**
- [x] commit messages are meaningful and follow good commit message guidelines
- [x] tests have been updated / new tests has been added (if needed)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by aashishaggarwal123 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java [hunk 1]
```diff
@@ -0,0 +1,633 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder;
+import static uk.gov.hmcts.reform.orgrolemapping.domain.service.RequestMappingService.ROLE_ASSIGNMENTS_RESULTS_KEY;
+
+
+@RunWith(MockitoJUnitRunner.class)
+class DroolJudicialOfficeMappingSSCS extends DroolBase {
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java:25]: Its good to end with Test for any Class name for Tests

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java [hunk 2]
```diff
@@ -0,0 +1,633 @@
...
+        });
+        assertEquals("caseworker-sscs-judge", roleAssignments.get(0).getRoleName());
+        assertEquals("case-allocator", roleAssignments.get(1).getRoleName());
+        assertEquals("task-supervisor", roleAssignments.get(2).getRoleName());
+        assertEquals("hmcts-judiciary", roleAssignments.get(3).getRoleName());
+        assertEquals("Salaried", roleAssignments.get(0).getAttributes().get("contractType").asText());
+        assertEquals("Salaried", roleAssignments.get(1).getAttributes().get("contractType").asText());
+        assertEquals("Salaried", roleAssignments.get(2).getAttributes().get("contractType").asText());
+        assertEquals("Salaried", roleAssignments.get(3).getAttributes().get("contractType").asText());
+    }
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java:61]: Please asset jurisdiction, worktypes, authorisations, primarylocations as well

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java [hunk 3]
```diff
@@ -0,0 +1,633 @@
...
+import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder;
+import static uk.gov.hmcts.reform.orgrolemapping.domain.service.RequestMappingService.ROLE_ASSIGNMENTS_RESULTS_KEY;
+
+
+@RunWith(MockitoJUnitRunner.class)
+class DroolJudicialOfficeMappingSSCS extends DroolBase {
+
+    @Test
+    void shouldReturnTribunalJudgeSalariedRoles() {
+
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java:29]: write parametarised test by using @CSVParameter to verify similar functionalities

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialRoleMappingSSCS.java [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+        //assertion
+        assertFalse(roleAssignments.isEmpty());
+        assertEquals(1, roleAssignments.size());
+        assertEquals("caseworker-sscs-financial-feepaid",roleAssignments.get(0).getRoleName());
+        assertEquals(judicialOfficeHolders.stream().iterator().next().getUserId(),roleAssignments.get(0).getActorId());
+        assertEquals("Fee-Paid", roleAssignments.get(0).getAttributes().get("contractType").asText());
+
+    }
+
+
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialRoleMappingSSCS.java:418]: Can you please add test for 
1. Invalid authorisation(expired enddate) and valid appointment
2. Invalid authorisation(wrong servicecode) and valid appointment
3. Invalid authorisation(expired enddate) and Invalid appointment(wrong servicecode)
4. Valid appointmentRoles
5. Invalid appointmentRoles

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java [hunk 5]
```diff
@@ -0,0 +1,633 @@
...
+        assertEquals("hmcts-judiciary", roleAssignments.get(3).getRoleName());
+        assertEquals("Salaried", roleAssignments.get(0).getAttributes().get("contractType").asText());
+        assertEquals("Salaried", roleAssignments.get(1).getAttributes().get("contractType").asText());
+        assertEquals("Salaried", roleAssignments.get(2).getAttributes().get("contractType").asText());
+        assertEquals("Salaried", roleAssignments.get(3).getAttributes().get("contractType").asText());
+    }
+
+    @Test
+    void shouldReturnRegionalTribunalSalariedRoles() {
+
```
Comment (ConorH994) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java:65]: Can you look into using parameterised tests?

There should be examples of this elsewhere in other drools tests.

Search in code for @ParameterizedTest

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialMappingTest.java [hunk 6]
```diff
@@ -294,6 +294,7 @@ void shouldReturnFeePaidTribunalJudgeRoles_withMultipleBookings() throws IOExcep
         judicialBooking2.setLocationId(null);
         judicialBooking2.setBeginTime(ZonedDateTime.now().minusDays(5));
         judicialBookings = Set.of(judicialBooking, judicialBooking2);
+        System.out.println(judicialBooking);
         //Execute Kie session
```
[the remark this answers is not in the document]
Comment (vladkovalskyi) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialMappingTest.java:298]: Removed.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java (hunk 1 above)
Comment (vladkovalskyi) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java:25]: Refactored class names.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java (hunk 2 above)
Comment (vladkovalskyi) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java:61]: Added the missing assertions.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java (hunk 3 above)
Comment (vladkovalskyi) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java]: Done

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java (hunk 5 above)
Comment (vladkovalskyi) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialOfficeMappingSSCS.java]: Done.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialRoleMappingSscsTest.java [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static uk.gov.hmcts.reform.orgrolemapping.domain.service.RequestMappingService.ROLE_ASSIGNMENTS_RESULTS_KEY;
+
+@RunWith(MockitoJUnitRunner.class)
+class DroolJudicialRoleMappingSscsTest extends DroolBase {
+
+    //sscs_caseworker_sscs_judge_org_role
+    @Test
+    void shouldReturnPresidentRoles() {
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialRoleMappingSscsTest.java:22]: Please use Parametrised Test for similar tests

Comment (vladkovalskyi) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialRoleMappingSscsTest.java]: Done.

**Review by sankar-padakula — APPROVED** [review]

**Pull request merged**